File: javascript/171.excel-sheet-column-number.js
/*
 * @lc app=leetcode id=171 lang=javascript
 *
 * [171] Excel Sheet Column Number
 */

// @lc code=start
/**
 * @param {string} columnTitle
 * @return {number}
 */
var titleToNumber = function (columnTitle) {
  let columnNumber = 0;
  for (let i = 0; i < columnTitle.length; i++) {
    columnNumber +=
      (columnTitle[i].charCodeAt() - 64) *
      Math.pow(26, columnTitle.length - i - 1);
  }
  return columnNumber;
};
// @lc code=end

File: rust/171.excel-sheet-column-number.rs
/*
 * @lc app=leetcode id=171 lang=rust
 *
 * [171] Excel Sheet Column Number
 */

// @lc code=start
impl Solution {
    pub fn title_to_number(column_title: String) -> i32 {
        let (mut n, mut column_number) = (0, 0);
        for i in column_title.chars().rev() {
            column_number += ((i as i32) - 64) * 26_i32.pow(n);
            n += 1;
        }
        return column_number;
    }
}
// @lc code=end
